DynamicPoolAllocator.hpp

```diff
@@ -140,7 +149,7 @@ class DynamicPoolAllocator
     struct Block *next = prev ? prev->next : freeBlocks;
 
     // Check if prev and curr can be merged
-    if ( prev && prev->data + prev->size == curr->data && !curr->isHead ) {
+    if ( prev && prev->data + prev->size == curr->data ) {
```

Comment:
Is this literally all you needed to merge blocks?

Reply:
I changed many things then simplified/debugged and in the end, yes this was all I needed :-).

Reply:
isHead was being used to determine when a fragmented block (one block per device allocation) became merged and could be freed.  Any blocks that had a chain of greater than 1 meant that some portion of that block was split off still allocated to someone.

While removing the check for isHead will allow us to merge blocks together beyond the previously split blocks, identifying which blocks may be freed back to the system will become slightly more complicated.

I will maintain an list of blocks that were originally allocated from the system and will compare memory ranges of available merged blocks to determine what is available to be split out and freed back to the system when/if needed.  This will amount to walking through the originally allocated block list and checking each blocks range with what is in the freeblock list.